python-frontend/src/main/java/org/sonar/python/cfg/ControlFlowGraphBuilder.java

```diff
@@ -186,13 +183,28 @@ private PythonCfgBlock build(Statement statement, PythonCfgBlock currentBlock) {
         return buildBreakStatement((BreakStatement) statement, currentBlock);
       case MATCH_STMT:
         return buildMatchStatement((MatchStatement) statement, currentBlock);
+      case FUNCDEF:
+        return buildFuncDefStatement((FunctionDef) statement, currentBlock);
       default:
         currentBlock.addElement(statement);
     }
 
     return currentBlock;
   }
 
+  private PythonCfgBlock buildClassDefStatement(ClassDef classDef, PythonCfgBlock successor) {
```

Comment:
Isn't this a bit misleading to have the parameter called `successor`?
It seems it is the `currentBlock` that is passed as an argument, and this is indeed the one we want to add the elements to.